spyder/plugins/editor/lsp/transport/common/producer.py

```diff
@@ -0,0 +1,114 @@
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# Copyright (c) 2009- Spyder Project Contributors
+#
+# Distributed under the terms of the MIT License
+# (see spyder/__init__.py for details)
+# -----------------------------------------------------------------------------
+
+
+"""
+Spyder MS Language Server Protocol v3.0 transport proxy base implementation.
+
+This module provides the base class from which each transport-specific client
+should inherit from. LanguageServerClient implements functions to handle
+incoming requests from the actual Spyder LSP client ZMQ queue and to
+encapsulate them into valid JSONRPC messages before sending them to the
+LSP server, using the specific transport mode.
+"""
+
+# Standard library imports
+import json
+import logging
+
+# Third party imports
+import zmq
+
+TIMEOUT = 5000
+
+logger = logging.getLogger(__name__)
+
+
+class LanguageServerClient:
```

Comment:
Please make this class inherit from `object` to make it a new class in Python 2. That should fix the errors on it.